runtime/compiler/env/j9method.cpp

```diff
@@ -2368,1502 +2369,1558 @@ TR_ResolvedJ9Method::TR_ResolvedJ9Method(TR_OpaqueMethodBlock * aMethod, TR_Fron
       _jniTargetAddress = NULL;
       }
 
-#define x(a, b, c) a, sizeof(b) - 1, b, (int16_t)strlen(c), c
-
    struct X
       {
+      typedef int8_t javaVersion_t;
+      #define AllPastJavaVer (std::numeric_limits<int8_t>::min())
+      #define AllFutureJavaVer (std::numeric_limits<int8_t>::max())
+
+      /**
+       * Constructor for recognized method specifying only the kind of
+       * method.  Typically used to mark an unknown placeholder entry.
+       */
+      X(TR::RecognizedMethod methodEnum) :
```

Comment:
@hzongaro I am worried this will have a negative affect on startup. The use of the function macro to expand the entries for the array will have resulted in a big blob of binary data being generated into the DLL which would be mapped into memory when the JIT is loaded. Converting this to a constructor call means, I think, that we have to run code to init each entry of the recognized methods table. If we have to run code that will likely take longer (potentially much longer) than mapping the bytes into memory directly. This could show up as a startup regression. Why do we need to convert to a constructor and not just add the appropriate entries to the initializes?

Reply:
Thanks for calling that out Andrew, @andrewcraik.  I didn't realize that could have such an impact on start-up performance.

My goal had been to make the changes more readable - the way wildcards are handled is a little bit ugly, and I was worried about introducing a whole lot more ugliness with native ranges.

I had started off by using the old `x` macro for the recognized methods that are not expected to be native (or are "don't care/unknown"), so those lines would be unchanged, and another macro for those expected to be native.  I can revert to that -- or possibly three for bounded and unbounded ranges for native methods -- and introduce another to clean-up the handling of wildcards.

Reply:
Andrew @andrewcraik, I've reworked this to restore the use of macros for aggregate initialization of the tables of recognized methods.